standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -0,0 +1,451 @@
+#[derive(Serialize, Deserialize, Debug, Clone, Encode, Decode)]
+pub struct SendLottery {
+    round_id: u32,
+    chain_id: u8,
+    token_id: Vec<u8>,
+    tx: Vec<u8>,
+    sequence: SequenceType,
+}
+#[derive(Serialize, Deserialize, Debug, Clone, Encode, Decode)]
+pub struct SendLotteryData {
+    data: SendLottery,
+    signature: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Command {
+    SubmitUtxo {
+        round_id: u32,
+        address: String,
+        utxo: (Txid, u32, u64),
+    },
+    SetAdmin {
+        new_admin: String,
+    },
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Error {
+    InvalidRequest,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Request {
+    GetAllRounds,
+    GetRoundInfo { round_id: u32 },
+    QueryUtxo { round_id: u32 },
+    // GetSignedTx { round_id: u32 },
+    PendingLotteryEgress { sequence: SequenceType },
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Response {
+    GetAllRounds {
+        round_id: u32,
+    },
+    GetRoundInfo {
+        token_number: u32,
+        winner_count: u32,
+    },
+    QueryUtxo {
+        utxo: BTreeMap<Address, (Txid, u32, u64)>,
+    },
+    // GetSignedTx { round_id: u32 },
+    PendingLotteryEgress {
+        lottery_queue_b64: String,
+    },
+    Error(Error),
+}
+
+impl BtcLottery {
+    /// Initializes the contract
+    pub fn new(secret: Option<ecdsa::Pair>) -> Self {
+        let token_set = BTreeMap::<u32, Vec<String>>::new();
+        let lottery_set = BTreeMap::<u32, BTreeMap<String, PrivateKey>>::new();
+        let utxo = BTreeMap::<u32, BTreeMap<Address, (Txid, u32, u64)>>::new();
+        let admin = AccountIdWrapper::from_hex(ALICE);
+        BtcLottery {
+            round_id: 0,
+            token_set,
+            lottery_set,
+            sequence: 0,
+            queue: Vec::new(),
+            secret,
+            utxo,
+            admin,
+        }
+    }
+
+    pub fn sign(
+        secp: &Secp256k1<All>,
+        digest: &[u8],
+        key: &PrivateKey,
+    ) -> Result<Signature, bitcoin::secp256k1::Error> {
+        Ok(secp.sign(&Message::from_slice(digest)?, &key.key))
+    }
+
+    pub fn new_round(&mut self, round_id: u32, total_count: u32, winner_count: u32) {
+        if !self.token_set.contains_key(&round_id) && !self.lottery_set.contains_key(&round_id) {
+            let secret = match self.secret.as_ref() {
+                Some(s) => s.to_raw_vec(),
+                None => {
+                    error!("LotteryNewRound: empty secret key");
+                    return;
+                }
+            };
+            let token_id: U256 = U256::from(round_id) << 128;
+            let mut round_token = Vec::new();
+            for index_id in 1..=total_count {
+                let nft_id = (token_id + index_id) | *TYPE_NF_BIT;
+                let token_id = format!("{:#x}", nft_id);
+                round_token.push(token_id);
+            }
```

Comment:
Let's not mix the names.

```
round_id: the number of the round, starting from 1
token_no: the number of the token in a round, starting from 1 to batch_size
token_id = (round_id << 128) | token_no | *TYPE_NF_BIT
```

Reply:
Now I use `token_round_id` to represent the high bits of token's round_id, and `token_no` to represent the number of the token in a round, and `token_id` for its true id

Reply:
I don't think `token_id` is a good name to represent the rounds, because it's the term in ERC1155